File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heap = []
    for i in scoville:
        heapq.heappush(heap,i)
    
    while heap:
        answer += 1
        
        if len(heap) >= 2:
            a1 = heapq.heappop(heap)
            a2 = heapq.heappop(heap) * 2
            heapq.heappush(heap,a1 + a2)
            if heap[0] >= K:
                break
        else:
            answer = -1
            break
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int answer = 0;
        int y = 0;
        int x1 = 0;
        int x2 = 0;
        
        for(int i=0; i<scoville.length; i++) {
            pq.offer(scoville[i]);
        }
        
        if(pq.peek() >= K) {
            return 0;
        }
        
        while(pq.size() >= 2) {
            x1 = pq.poll();
            x2 = pq.poll();
            y = x1 + (x2 * 2);
            pq.offer(y);
            answer ++;
            
            if(pq.peek() >= K) {
                return answer;
            }
        }
        
        return -1;
    }
}